Duplicati/Server/webroot/ngax/scripts/directives/backupEditUri.js

```diff
@@ -245,6 +253,82 @@ backupApp.directive('backupEditUri', function(gettextCatalog) {
             builduri(performConnectionTest);
         };
 
+        function perform_oauth_start_token_creation (uri) {
+            const ur = new URL(uri)
+            // check for the oAuth server in the backend specific parameters
+            var oauthUrl = ur.searchParams.get("oauth-url");
+            if (oauthUrl != null && oauthUrl.length > 0) {
+                if (oauthUrl.endsWith("/refresh"))
+                    oauthUrl = oauthUrl.substr(0, oauthUrl.length - 7);
+                $scope.oauth_service_link = oauthUrl;
+            }
```

Comment:
The reason why it doesn't work with advanced options on step 5 is that this only uses the options in the backend uri. I think that is fine since test connection should only look at that part of the options. I believe any other option in step 5 would also be ignored for the connection test. Most people (of the few that need a different oauth handler) would use a global option anyways.